src/prefect/task_engine.py

```diff
@@ -696,11 +696,16 @@ def initialize_run(
                         self.logger.debug(
                             f"Created task run {self.task_run.name!r} for task {self.task.name!r}"
                         )
-                        labels = (
-                            flow_run_context.flow_run.labels if flow_run_context else {}
-                        )
+
+                        parent_labels = {}
+                        if parent_flow_run_context and parent_flow_run_context.flow_run:
+                            parent_labels = parent_flow_run_context.flow_run.labels
```

Comment:
Should we be getting labels from parent task runs too?

Reply:
In a separate PR we will be rewriting which labels gets passed down since the parent span already includes label information, we don't need to include it in each of the child spans and in many cases it could overwrite 'real' data or attributes set by users.